### PR TITLE
BXSMPROD-685: upgrade of weld to 3.x

### PR DESCRIPTION
**Pull request by mareknovotny**
https://issues.redhat.com/browse/BXMSPROD-685

Parent PR https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1320

### CODE REVIEW

**Comment by mareknovotny** [comment]
Enforcer failures, will fix them

**Comment by mareknovotny** [comment]
Jenkins retest this please

the build was aborted as it went over 4h 30 min

File: jbpm-audit/pom.xml [hunk 1]
```diff
@@ -106,7 +123,7 @@
       <artifactId>assertj-core</artifactId>
       <scope>test</scope>
     </dependency>
-    <dependency>
+    <!--dependency>
```
[the remark this answers is not in the document]
Comment (mareknovotny) [reply jbpm-audit/pom.xml:126]: this is draft so the comments are there just for quick turnaround if it doesn't work ;)

[the remark this answers is not in the document]
Comment (gmunozfe) [reply jbpm-audit/pom.xml:126]: ok, sorry, didn't realize it is still a draft

**Comment by mareknovotny** [comment]
I have still some doubts about com.sun.xml.bin:jaxb-impl and jaxb-core where there are comments referencing declaration due JDK5 and also JDK11 deps are there in default build reactor instead of in automaticaly activated profile when jdk 11 is used, but this is not related to this Weld and other deps upgrades so keeping that a side ;)

**Comment by mareknovotny** [comment]
@gmunozfe @elguardian any clue if `org.jbpm.test.functional.jobexec.AsyncTaskTest.testTaskErrorHandling` is failing due this changes?

**Comment by elguardian** [comment]
@mareknovotny sry for deleyed responsed... I think it is because of the jaxb stuff.

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by mareknovotny** [comment]
Jenkins retest this please

**Review by yesamer — APPROVED** [review]

**Pull request merged**